---
title: 架构设计引发的血案
cover: http://t-blog-images.aijs.top/img/bug.webp
categories: bug
tags: [bug]
date: 2022-04-29 10:56:00
---
## 记录个bug,并在此文中规避
此处记下 hexo-theme-aurora 源码bug，没空时候帮他改
```
.article .article-content p {
    font-size: .875rem;
    line-height: 1.25rem;
    margin-bottom: 0.5rem;
    padding-bottom: 1rem;
    word-break: break-all;
}
```
## bug复现
![](http://t-blog-images.aijs.top/img/20220429103417.png)
![](http://t-blog-images.aijs.top/img/20220429103946.png)
```js
(node:29) UnhandledPromiseRejectionWarning: Error: Forbidden
    at /code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:13141:29
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Object.getDataReport (/code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:11494:19)
    at async Promise.all (index 3)
    at async Function.getInitialProps (/code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:12208:22)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:29) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 19)
(node:29) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:29) UnhandledPromiseRejectionWarning: Error: Forbidden
    at /code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:13141:29
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Object.getDataReport (/code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:11494:19)
    at async Promise.all (index 3)
    at async Function.getInitialProps (/code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:12208:22)
(node:29) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 44)
Error: invalid json response body at https://login-daily.敏感数据-inc.cn:7799/error/illegal/location reason: Unexpected token < in JSON at position 0
    at /code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:13141:29
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Object.refreshPendingPerformanceNum (/code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:11487:25)
    at async Promise.all (index 2)
    at async Function.getInitialProps (/code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:12208:22)
Error: invalid json response body at https://login-daily.敏感数据-inc.cn:7799/error/illegal/location reason: Unexpected token < in JSON at position 0
    at /code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:13141:29
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Promise.all (index 0)
    at async Function.getInitialProps (/code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:12208:22)
Error: Forbidden
    at /code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:13141:29
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Object.getDataReport (/code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:11494:19)
    at async Promise.all (index 3)
    at async Function.getInitialProps (/code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:12208:22)
Error: Forbidden
    at /code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:13141:29
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Promise.all (index 0)
    at async Function.getInitialProps (/code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:12208:22)
Error: invalid json response body at https://login-daily.敏感数据-inc.cn:7799/error/illegal/location reason: Unexpected token < in JSON at position 0
    at /code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:13141:29
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Promise.all (index 0)
    at async Function.getInitialProps (/code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:12208:22)
Error: invalid json response body at https://login-daily.敏感数据-inc.cn:7799/error/illegal/location reason: Unexpected token < in JSON at position 0
    at /code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:13141:29
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Object.refreshPendingPerformanceNum (/code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:11487:25)
    at async Promise.all (index 2)
    at async Function.getInitialProps (/code/client/.next/server/pages/_app-a053507d8512cb10a3ff.js:12208:22)

```

## 源代码

```tsx
const [resp] = await Promise.all([
  getLoginInfo(ctx),
  CommonModel.effects.refreshPendingObjectiveNum(req, ctx),
  CommonModel.effects.refreshPendingPerformanceNum(req, ctx),
  CommonModel.effects.getDataReport(req, ctx),
])

if (resp.success) {
  UserModal.actions.setUser(resp.result, req)
}
```
## 增加异常捕获
```tsx
const [resp] = await Promise.all([
  getLoginInfo(ctx),
  CommonModel.effects.refreshPendingObjectiveNum(req, ctx),
  CommonModel.effects.refreshPendingPerformanceNum(req, ctx),
  CommonModel.effects.getDataReport(req, ctx),
]).catch(e=>{
  console.log("发生错误")
  console.log(e)
})
```
## 日志
日志仅 `Forbidden` 有效信息，大概是接口
```tsx
发生错误
Error: Forbidden
    at /code/client/.next/server/pages/_app-7be1c2d9b56a41e863e3.js:13144:29
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Promise.all (index 0)
    at async Function.getInitialProps (/code/client/.next/server/pages/_app-7be1c2d9b56a41e863e3.js:12208:22)

```


## 总结：
1. 因为acl 企业权限整了180天（顶配），时不时就出现系统瘫痪的情况，`日常`、`线上`都有此问题这架构设计的，估计没谁了。完全可以改成离职后，系统自动将用户禁权即可
2. 除此之外，sso也有个通病，三个月没改过密码，用户账户无法登录，需要重新密码方可登录